scripts/enable-ecs-agent-gpu-support.sh

```diff
@@ -1,7 +1,17 @@
 #!/usr/bin/env bash
 set -ex
 
-if [[ $AMI_TYPE != "al2gpu" && $AMI_TYPE != "al2keplergpu" ]]; then
+# Makes sure that a compatible version of gcc is used for compiling NVIDIA driver.
+set_compatible_gcc_version_for_nvidia_compile() {
+    # Currently a compatible version of gcc is assumed to be used by default, unless the AMI recipe uses kernel 5.10.
+    if [[ $AMI_TYPE == *"kernel5dot10gpu" ]]; then
+        # Explicitly use gcc10 since gcc version for compiling the NVIDIA driver must match gcc version with which the
+        # Linux kernel was compiled.
+        sudo sed -i "s/'make' -j2 module/& CC=\/usr\/bin\/gcc10-cc/" /usr/src/${MODULE_NAME}-${MODULE_VERSION}/dkms.conf
+    fi
+}
+
+if [[ $AMI_TYPE != "al2"*"gpu" ]]; then
```

Comment:
Do we need this check still? It looks like we're only explicitly setting a gcc version if the AMI is `al2kernel5dot10gpu` and for all other cases we'll just exit with a 0 code.